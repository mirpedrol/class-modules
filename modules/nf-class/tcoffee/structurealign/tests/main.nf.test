nextflow_process {

    name "Test Process TCOFFEE_STRUCTUREALIGN"
    script "../main.nf"
    process "TCOFFEE_STRUCTUREALIGN"

    tag "modules"
    tag "modules_nfclass"
    tag "tcoffee"
    tag "tcoffee/structurealign"
    tag "untar"

    test("fasta - align_with_structure") {

        setup {

            run("UNTAR") {
                script "../../../untar/main.nf"
                process {
                    """
                    input[0] = [ [ id:'test' ],
                                 file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/structures/seatoxin-ref.tar.gz", checkIfExists: true)
                               ]
                    """
                }
            }
        }

        when {
            process {
                """
                input[0] = [ [ id:'test' ],
                             file("https://raw.githubusercontent.com/nf-core/test-datasets/multiplesequencealign/testdata/setoxin-ref.fa", checkIfExists: true)
                           ]
                input[1] = UNTAR.out.untar.map { meta,dir -> [[ id:'test' ], [] ,file(dir).listFiles().collect()]}
                """

            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert path(process.out.alignment.get(0).get(1)).getTextGzip().contains("1ahl") },
                { assert snapshot(process.out.versions).match("versions_structure") }
            )
        }

    }
}