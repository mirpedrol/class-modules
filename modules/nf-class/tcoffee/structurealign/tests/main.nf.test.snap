{
    "versions_structure": {
        "content": [
            [
                "versions.yml:md5,dd47f1490a2c609b1a60a6558f8468ee"
            ]
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.2"
        },
        "timestamp": "2024-07-11T09:17:13.750477015"
    }
}